src/CodeGen/ODataT4CodeGenerator.cs

```diff
@@ -425,6 +440,38 @@ public void ValidateAndSetIgnoreUnexpectedElementsAndAttributesFromString(string
     this.IgnoreUnexpectedElementsAndAttributes = boolValue;
 }
 
+/// <summary>
+/// Set the CustomHttpHeaders property from the custom http headers string.
+/// </summary>
+/// <param name="stringValue">The value to set.</param>
+public void ValidateAndSetCustomHttpHeadersFromString(string headersValue) 
+{
+    if (String.IsNullOrWhiteSpace(headersValue))
+    {
+        return;
+    }
+
+    this.CustomHttpHeaders = new List<string>();
+
+    string[] headerElements = headersValue.Split(new string[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);
+    foreach (var headerElement in headerElements)
+    {
+        // Trim header for empty spaces
+        var header = headerElement.Trim();
+
+        // A header string must have atleast one colon
+        if (header.Contains(':'))
+        {
+            CustomHttpHeaders.Add(header);
+        }
+        else
+        {
+            throw new ArgumentException("A http header string must have a colon delimeter");

```

Comment:
```suggestion
            throw new ArgumentException($"The http header string {header} does not have a colon separator");

```

Reply:
Considering a header string can contain any character and spaces, adding it to the error message can be very messy

Reply:
Not sure I see why that is an issue - unless a header is really long and complicated, which is unlikely (you could also truncate it if you're worried about the length). The reason I propose this approach is to make the error message more informative by giving  hint and the dev immediately knows which header is erroneous instead of hunting for it